src/tool/subcommands/snapshot_cmd.rs

```diff
@@ -140,6 +157,26 @@ impl SnapshotCommands {
                 )
                 .await
             }
+            Self::Validate {
+                check_links,
+                check_network,
+                check_stateroots,
+                snapshot_files,
+            } => {
+                for file in snapshot_files {
```

Comment:
Idea: This will stop validation at the first invalid file. The subcommand would be more flexible if it'd go through all the provided snapshots (bonus points if in parallel, though I'd expect some stdout thread races) and then list valid and invalid snapshots, returning an error if any was corrupted. We could also add a flag `--fail-fast` to have the same behaviour as now.

What do you think?

Reply:
@LesnyRumcajs PTAL. And I'm having trouble handling the stdout race in parallel, as you mentioned. 😄

Reply:
You can get around this by changing `validate_with_blockstore` to accept a `Writer`. You'd pass it, e.g., a string buffer to be printed once the operation is done.

Reply:
Thank you. I have two questions here.
- Do you mean to provide a distinct string buffer for every snapshot?
- `Validate_with_blockstore` uses a ProgressBar, string buffer seems not an option.

Reply:
1. Yeah, distinct buffer. The entire text shouldn't be long anyway.
2. That does seem a bit challenging. We could either do some terminal splitting magic (hardcore-level, quite bug-prone) or just allow disabling the progress bar.

That said, given it's not trivial to implement, it will be more pragmatic to settle with the sequential validation and revisit the parallel one only when it's actually needed.